WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -475,15 +475,15 @@ public async Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(Connect
 				}
 			}
 
-			var amountZeroTask = round.AmountCredentialIssuer.PrepareResponse(request.ZeroAmountCredentialRequests, cancellationToken);
-			var vsizeZeroTask = round.VsizeCredentialIssuer.PrepareResponse(request.ZeroVsizeCredentialRequests, cancellationToken);
-			Task<ICommitableCredentialsResponse>? amountRealTask = null;
-			Task<ICommitableCredentialsResponse>? vsizeRealTask = null;
+			var amountZeroTask = round.AmountCredentialIssuer.HandleRequest(request.ZeroAmountCredentialRequests, cancellationToken);
+			var vsizeZeroTask = round.VsizeCredentialIssuer.HandleRequest(request.ZeroVsizeCredentialRequests, cancellationToken);
+			Task<CredentialsResponse>? amountRealTask = null;
+			Task<CredentialsResponse>? vsizeRealTask = null;
 
 			if (round.Phase is Phase.ConnectionConfirmation)
 			{
-				amountRealTask = round.AmountCredentialIssuer.PrepareResponse(realAmountCredentialRequests, cancellationToken);
-				vsizeRealTask = round.VsizeCredentialIssuer.PrepareResponse(realVsizeCredentialRequests, cancellationToken);
+				amountRealTask = round.AmountCredentialIssuer.HandleRequest(realAmountCredentialRequests, cancellationToken);
+				vsizeRealTask = round.VsizeCredentialIssuer.HandleRequest(realVsizeCredentialRequests, cancellationToken);
 			}
 
 			using (await AsyncLock.LockAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
What if we cannot get the lock here in time? Shouldn't the client try again? Should it be banned instantly?

Reply:
no, it should only be banned if the crypto error is thrown from one of the tasks

Reply:
How would it look like?

Reply:
sorry i misunderstood the comment, if the lock acquisition times out, that's like a 503 error code, but afaik the only way in which it can do that is if the cancellation token triggers, which implies some the client connection closed anyway

it doesn't make sense to retry the lock but it does make sense to cache the response for idempotency reasons (@lontivero yet another argument for going with a simple manual collection instead of fancy middleware)

Reply:
I am thinking about if it is safe to change the Prepare/Commit concept. It was robust and clean. With this PR it is not clear to me, if is safe to fail and what happens with the used credentials. We want to ensure this cannot go wrong with Banning and Idempotency?